trafficFlow/carFollowingModel/drivers/intelligentDriver.py

```diff
@@ -38,28 +38,69 @@ class IntelligentDriver(BaseDriver):
         according to the intelligent driver model
     """
 
-    def __init__(self, s_0, v_0, delta, T, a, b, length=4.):
-        super().__init__()
+    def __init__(self, s_0, v_0, delta, T, a, b, length=4., lane_change_safety_distance=50., label=""):
```

Comment:
A safety distance of 50 seems to be too high if the road is only 300.

Reply:
This points to a general problem which I was not able to solve so far:
If I reduce the safety distance to let's say 20 meters, cars sometimes do not change to the left lane although it is free. This is because the cars slow down, as given by the acceleration in the intelligent driver model, and do not reach a distance to the predecessor that is below the safety distance. That's why the cars do not change. I tried to introduce another condition which gives a car the opportunity to change to the left lane if the car in front is not as fast as the car itself wants to be (and all other conditions like safety distance to cars on the left lane are fulfilled). But that did not work so far.

Reply:
Yep, there should probably be another condition, the safety distance should mainly be used for the cars in the lane the vehicle switches to, and the decision to switch should be independent. The condition you tried seems to make sense, maybe it will work once the other issues are fixed.

Reply:
You're right, I will first fix the other issues and then have a look at the condition when to change the lane.